Doc/library/multiprocessing.shared_memory.rst

```diff
@@ -64,26 +64,38 @@ copying of data.
    memory block may be larger or equal to the size requested.  When attaching
    to an existing shared memory block, the ``size`` parameter is ignored.
 
+   *track*, when enabled, registers the shared memory block with a resource
+   tracker process.  This process ensures proper cleanup of shared memory
+   blocks even when all other processes with access to the memory have failed
+   to do so (mainly due to being killed by signals).  Unless a Python process
+   was created using any of the :mod:`multiprocessing` facilities (such as
+   :class:`multiprocessing.Process`), it will receive its own resource tracker
+   process when accessing shared memory with *track* enabled.  This will cause
+   the shared memory to be deleted by the resource tracker of the first
+   process that terminates.  To avoid this issue, users of :mod:`subprocess` or
+   standalone Python processes should set *track* to ``False`` when there is
+   already another process in place that does the bookkeeping.  *track* has
+   an effect only on POSIX.  Windows has its own tracking and does not use the
+   resource tracker.
```

Comment:
These explanations feel long-winded and not very helpful at the same time. How about something like this:

```suggestion
   to do so (mainly due to being killed by signals). As long as Python processes
   are spawned from a common ancestor using :mod:`multiprocessing` facilities,
   they share a common resource tracker process, and lifetime of shared memory
   segments is handled automatically among those processes. However,
   this mechanism becomes undesirable if a shared memory segment is also
   used by unrelated processes, for example if the shared memory segment name
   is given on the command line to another program. In this case, *track*
   should be set to ``False`` and the application should find other ways to
   delete a shared memory segment that is not in use anymore.

   On Windows, *track* is ignored as the operating system transparently
   deletes shared memory when all the processes referencing it are gone.
```

Reply:
Hmm, that's a fair point. How about the following then:
```suggestion
   to do so (mainly due to being killed by signals). As long as Python processes
   are spawned from a common ancestor using :mod:`multiprocessing` facilities,
   they share a common resource tracker process, and lifetime of shared memory
   segments is handled automatically among those processes.
   
   However, this automatic resource cleanup can become undesirable in
   other situations, such as:

   * if a shared memory segment is used by a Python process that is not
     part of the same :mod:`multiprocessing` process tree;
   * if a shared memory segment is used by a Python process that was not
     launched using :mod:`multiprocessing`;
   * if a shared memory segment is used by a non-Python process.
   
   In such situations, it is recommended that *track* is set to ``False`` and
   that the application implement its own mechanism to handle the lifetime
   of shared memory segments.

   On Windows, *track* is ignored as the operating system transparently
   deletes shared memory when all the processes referencing it are gone.
```

Reply:
I've taken your first and last sentence mostly unchanged and put some of the original things in between. I think subprocess is worth mentioning once.